### PR TITLE
Fix TypeError crash in SEP-7 `replace` parameter parsing

### PR DESCRIPTION
**Pull request by CassioMG**
### Summary

- Fix unhandled `TypeError` in `sep7ReplacementsFromString` when the `replace` parameter string is missing the `;` hints delimiter
- Add spec-compliant validation that rejects `replace` values with unbalanced reference identifiers
- Add test coverage for invalid replacement strings and edge cases

### Problem

`sep7ReplacementsFromString` calls `.split()` on `hintsString` without checking if it exists. When a `replace` parameter value lacks the `;` delimiter (e.g. `sourceAccount:X` instead of `sourceAccount:X;X:some hint`), the destructuring on line 165 leaves `hintsString` as `undefined`, and the immediate `.split()` call on line 166 throws:

```
TypeError: Cannot read properties of undefined (reading 'split')
```

### Fix

Per the [SEP-7 spec](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0007.md#operation-tx):

> The `replace` value is considered invalid unless the `reference_identifier`s are balanced on both sides of the separating semicolon.

The fix:
1. Guards against `undefined` on `hintsString` to prevent the `TypeError`
2. Validates that reference identifiers are balanced between the txrep side and the hints side
3. Throws `Sep7InvalidUriError` with a descriptive message when identifiers are unbalanced, instead of silently accepting invalid input

### Tests added

- Missing hints section (no `;` delimiter) — single and multiple fields — throws `Sep7InvalidUriError`
- Unbalanced identifiers (spec example: `sourceAccount:X;Y:The account`) — throws `Sep7InvalidUriError`
- `undefined` and empty string input — returns `[]`
- End-to-end: `Sep7Tx.getReplacements()` with a URI whose `replace` param has no hints — throws `Sep7InvalidUriError`
- Existing tests for valid replacement strings continue to pass

### CODE REVIEW

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts [hunk 1]
```diff
@@ -163,13 +163,15 @@ export const sep7ReplacementsFromString = (
...
 
-  hintsList
-    .map((item) => item.split(ID_DELIMITER))
-    .forEach(([id, hint]) => (hintsMap[id] = hint));
+  if (hintsString) {
+    const hintsList = hintsString.split(LIST_DELIMITER);
+    hintsList
+      .map((item) => item.split(ID_DELIMITER))
+      .forEach(([id, hint]) => (hintsMap[id] = hint));
+  }
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts:202]: the hint is actually required as per Sep-7 spec so we are throwing an error now instead of silently accepting it: https://github.com/stellar/typescript-wallet-sdk/pull/227/commits/5a3bd1e4755ea58a93f744131e42df929d7dccd3

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts [hunk 2]
```diff
@@ -162,16 +162,49 @@ export const sep7ReplacementsFromString = (
...
-  const hintsMap: { [id: string]: string } = {};
+  const txrepList = txrepString.split(LIST_DELIMITER);
+  const txrepIds: string[] = [];
+  txrepList.forEach((item) => {
+    const parts = item.split(ID_DELIMITER);
+    if (parts.length < 2 || !parts[1]) {
+      throw new Sep7InvalidUriError(
+        "the 'replace' parameter has an entry missing a reference identifier",
+      );
+    }
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts:182]: https://github.com/stellar/typescript-wallet-sdk/pull/227/commits/ed5cfab008115517880dc03633ca530406504273

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts [hunk 3]
```diff
@@ -162,16 +162,49 @@ export const sep7ReplacementsFromString = (
     return [];
   }
 
+  if (replacements.length > 4096) {
+    throw new Sep7InvalidUriError(
+      "the 'replace' parameter exceeds the maximum allowed length",
+    );
+  }
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts]: https://github.com/stellar/typescript-wallet-sdk/pull/227/commits/ed5cfab008115517880dc03633ca530406504273

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts [hunk 4]
```diff
@@ -162,16 +162,49 @@ export const sep7ReplacementsFromString = (
...
-    .map((item) => item.split(ID_DELIMITER))
-    .forEach(([id, hint]) => (hintsMap[id] = hint));
+  const hintsMap = Object.create(null) as Record<string, string>;
 
-  const txrepList = txrepString.split(LIST_DELIMITER);
+  if (hintsString) {
+    const hintsList = hintsString.split(LIST_DELIMITER);
+    hintsList
+      .map((item) => item.split(ID_DELIMITER))
+      .forEach(([id, hint]) => (hintsMap[id] = hint));
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts]: https://github.com/stellar/typescript-wallet-sdk/pull/227/commits/ed5cfab008115517880dc03633ca530406504273

File: @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts [hunk 5]
```diff
@@ -162,16 +163,55 @@ export const sep7ReplacementsFromString = (
     return [];
   }
 
+  if (replacements.length > URI_REPLACE_MAX_LENGTH) {
+    throw new Sep7InvalidUriError(
+      "the 'replace' parameter exceeds the maximum allowed length",
+    );
+  }
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts:170]: https://github.com/stellar/typescript-wallet-sdk/pull/227/changes/737c0ba27bb33c2057a55b21fc444cea375a59b6
